### DISCUSSION TITLE
USB Host Mode on Raspberry Pi PicoW.

### DISCUSSION THREAD

**Discussion by Mohanbandaru16** [micropython/micropython#15394] [RP2040 / Pico]
Hi everyone,

I'm working on a project where I need to communicate with an amplifier device using a Raspberry Pi Pico W via USB, but I'm not sure where to start. Could anyone please provide some guidance? (I have already attempted this using the Pico SDK and TinyUSB in C but failed to do so in MicroPython.)

Thanks.

**Comment by Josverl** [comment]
Firstly the Micropython USB docs explicitly state :
> Currently only USB device is implemented, not USB host.

I would start looking for what types of  input that Amp can process.
there are many different ways to transfer Audio over USB, see : https://www.usb.org/documents?search=USB+AUdio
AFAIK USB Audio is not classified as a USB HID protocol - and so the new USB HID drivers may not help you here 

There is also I2S over USB - and I2S is available on rp2040 , but id not start there unless you are sure that the Amp can handle that,


Update : Just found this Thread on USB Audio interface #10842

**Comment by Mohanbandaru16** [comment]
Thanks for your reply. For now ,i will wait for Micro python to give USB host mode support. until then i will stick to Pico SDK.
